lab_1_keywords_tfidf/main.py

```diff
@@ -2,22 +2,35 @@
 Lab 1
 Extract keywords based on frequency related metrics
 """
-from typing import Optional, Union
+from typing import Optional, Union, List
+import re
 
 
-def clean_and_tokenize(text: str) -> Optional[list[str]]:
+
+def clean_and_tokenize(text: str) -> Optional[List[str]]:
     """
-    Removes punctuation, casts to lowercase, splits into tokens
+      Removes punctuation, casts to lowercase, splits into tokens
+      Parameters:
+      text (str): Original text
+      Returns:
+      list[str]: A sequence of lowercase tokens with no punctuation
+      In case of corrupt input arguments, None is returned
+      """
+    pass
+
+    if isinstance(text, str):
+        text = re.sub('[!@#$\n-.,]', '', text)
+        text = text.replace('  ', ' ')
+        text.strip()
+        text = text.lower()
+        text = text.split()
+        return text
+    else:
```

Comment:
it is better to change the order of branches here 
```
if <condition>:
    <shorter block of code>
else:
   <longer  block of code>
```
This way code will become more readable

Reply:
Done.